app/index.js

```diff
@@ -1,13 +1,26 @@
 const express = require("express");
 const handleAsync = require("@app/middleware/handle-async");
+const basicAuth = require("express-basic-auth");
 const session = require("@app/middleware/jwt-session");
 const auth = session.authenticate;
+const basicAuthCfg = require("@app/config").basicAuth;
 
 const app = express();
 
 app.use(require("helmet")());
 app.use(require("@app/middleware/logging"));
 
+if (basicAuthCfg.user && basicAuthCfg.password) {
+  app.use(
+    basicAuth({
+      users: {
+        [basicAuthCfg.user]: basicAuthCfg.password
+      },
+      challenge: true
+    })
+  );
+}
+
```

Comment:
What do you think about moving this into e.g. `app/middleware/basic-auth` so that this would become

```js
app.use(require("@app/middleware/basic-auth"));
```

?

I like the idea of `app/index.js` being just the "routes", with actual logic elsewhere. This keeps the routes easy to scan. What do you think?

Reply:
Yeah. fine with me.  i'll move it.